packages/widget-todo/tests/ChipInput.test.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { render, screen, within } from "@testing-library/react";
+import ChipInput from "../src/components/ChipInput";
+import userEvent from "@testing-library/user-event";
+
+const noop = () => undefined;
+
+test("renders label", async () => {
+  const label = "Add some tags!";
+  render(<ChipInput label={label} value={[]} onChange={noop} />);
+
+  expect(screen.getByText(label)).toBeInTheDocument();
+});
+
+test("renders tags", async () => {
+  const tags = ["tagOne", "tagTwo"];
+  render(<ChipInput value={tags} onChange={noop} />);
+
+  tags.forEach((tag) =>
+    expect(screen.getByRole("button", { name: tag })).toBeInTheDocument()
+  );
+});
+
+test("triggers `onChange` with new value when enter is pressed", async () => {
+  const onChange = jest.fn();
+  const oldTags = ["OldTag"];
+
+  render(<ChipInput value={oldTags} onChange={onChange} />);
+
+  const userInputText = "NewTag";
+  userEvent.type(screen.getByRole("textbox"), userInputText + "{enter}");
+
+  expect(onChange).toHaveBeenCalledWith([...oldTags, userInputText]);
+});
+
+test("Clicking the X removes a tag", async () => {
+  const onChange = jest.fn();
+  const tagToRemoveLabel = "tag-to-remove";
+  const oldTags = ["some-tag", tagToRemoveLabel, "other-tag"];
+
+  render(<ChipInput value={oldTags} onChange={onChange} />);
+
+  const tagToRemove = screen.getByRole("button", { name: tagToRemoveLabel });
+
+  userEvent.click(tagToRemove.querySelector("svg")!);
```

Comment:
Unfortunately the MUI Chip (and MUI generally) has not the greatest a11y (as [they admit themselves](https://github.com/mui/material-ui/issues/20470)), which makes it also harder to test at times. So I have to directly address the svg tag.

Interestingly in their own tag, [they swap out the default delete icon to test it](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Chip/Chip.test.js#L204-L237).